mxcube3/routes/Login.py

```diff
@@ -113,22 +139,107 @@ def signout():
     """
     Signout from Mxcube3 and reset the session
     """
+    if mxcube.CURRENTLY_MOUNTED_SAMPLE:
+        if mxcube.CURRENTLY_MOUNTED_SAMPLE.get('location', '') == 'Manual':
+            mxcube.CURRENTLY_MOUNTED_SAMPLE = ''
+
+    user = remove_user(session.sid)
+
+    if not logged_in_users(exclude_inhouse=False):
+        qutils.save_queue(session)
+        mxcube.queue = qutils.new_queue()
+        mxcube.shapes.clear_all()
+        qutils.init_queue_settings()
+        mxcube.SELECTED_PROPOSAL = None
+        mxcube.SELECTED_PROPOSAL_ID = None
+        clear_messages()
+        # ony clear hwobj if this was the last user
+        if hasattr(mxcube.session, 'clear_session'):
+            mxcube.session.clear_session()
+    session.clear()
+
+    logging.getLogger('HWR').info('[Login] %s user signout' % user['loginID'])
+
+    return make_response("", 200)
+
+@mxcube.route("/mxcube/api/v0.1/forcesignout")
+def forcesignout():
+    """
+    Force signout from Mxcube3 and reset the session
+    """
     qutils.save_queue(session)
     mxcube.queue = qutils.new_queue()
     mxcube.shapes.clear_all()
-    qutils.init_queue_settings()
-    if hasattr(mxcube.session, 'clear_session'):
-        mxcube.session.clear_session()
+    mxcube.session.clear_session()
 
     if mxcube.CURRENTLY_MOUNTED_SAMPLE:
         if mxcube.CURRENTLY_MOUNTED_SAMPLE.get('location', '') == 'Manual':
             mxcube.CURRENTLY_MOUNTED_SAMPLE = ''
 
-    user = remove_user(session.sid)
-    session.clear()
+    state_storage.flush()
+
+    mxcube.SELECTED_PROPOSAL = None
+    mxcube.SELECTED_PROPOSAL_ID = None
 
+    session.clear()
+    socketio.emit("signout", {}, namespace='/hwr')
     return make_response("", 200)
 
+@mxcube.route("/mxcube/api/v0.1/forceusersignout", methods=["POST"])
+def forceusersignout():
```

Comment:
Do you really want all this logic directly in the route an cant we reuse some logic from the logout routine ?